network/wsNetwork.go

```diff
@@ -1444,6 +1472,27 @@ func (wn *WebsocketNetwork) innerBroadcast(request broadcastRequest, prio bool,
 		if request.tags[i] != protocol.MsgDigestSkipTag && len(d) >= messageFilterSize {
 			digests[i] = crypto.Hash(mbytes)
 		}
+
+		if needCompressedData {
+			if request.tags[i] == protocol.ProposalPayloadTag {
```

Comment:
to simplify this you could introduce a new message tag CompressedProposalPayloadTag?